js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_6
+ * by Diane Zevenbergen
+ */
+
+describe('Enhanced Date', function () {
+    var setDate = enhancedDate.setDate;
+    var getDate = enhancedDate.getDate;
+    var getDayName = enhancedDate.getDayName;
+    var getMonthName = enhancedDate.getMonthName;
+    var isToday = enhancedDate.isToday;
+    var isFuture = enhancedDate.isFuture;
+        
+    // setDate tests
+    describe('setDate method', function () {
+
+        it('should exist', function () {
+            expect(setDate()).to.exist;
+        });
+
+        it('should be instanceOf Date', function () {
+            expect(setDate()).is.an.instanceOf(Date);
+        });
+        
+        // test .deep.equal with 5/14/2015
+        it('matching dates should be deep equal', function () {
+            expect(setDate(new Date(2015,4,14))).to.deep.equal(new Date('5/14/2015'));
+        });
+
+        // test that .deep.equal fails comparing 5/14/2015 to current day
+        it('different dates should not be deep equal', function () {
+            expect(setDate(new Date(2015,4,14))).to.not.deep.equal(new Date());
+        });
+    });
+    
+    // getDate tests
+    describe('getDate method', function () {
+        
+        it('should exist', function () {
+            expect(getDate()).to.exist;
+        });
+        
+        // test that passing 'true' returns date obj
+        it('passing "true" should return instanceOf Date', function () {
+            expect(getDate(true)).is.an.instanceOf(Date);
+        });
+        
+        // test that passing nothing returns number (Epoch ms)
+        it('passing nothing should return a number', function () {
+            expect(getDate()).to.be.a('Number');
+        });
+        
+        // QUESTION:  is something like this too goofy for testing?
```

Comment:
Hmm, I probably wouldn't set up a function in this manner simply because it makes the implementation of the test a little confusing. If you wanted to validate the return value of getTime() equalled the return value of getDate, I would simply call getTime on a Date instantiation using the same base date.